bindings/masm/keystone_x64.inc

```diff
@@ -0,0 +1,180 @@
+KS_ERR_ASM_LABEL_INVALID        EQU 162     ; invalid label
+KS_ERR_ASM_FRAGMENT_INVALID     EQU 163     ; invalid fragment
+
+; generic input assembly errors - architecture specific
+KS_ERR_ASM_INVALIDOPERAND       EQU KS_ERR_ASM_ARCH,
+KS_ERR_ASM_MISSINGFEATURE       EQU 513
+KS_ERR_ASM_MNEMONICFAIL         EQU 514
+
+; ks_err_asm_x86
+KS_ERR_ASM_X86_INVALIDOPERAND   EQU KS_ERR_ASM_ARCH
+KS_ERR_ASM_X86_MISSINGFEATURE   EQU KS_ERR_ASM_MISSINGFEATURE
+KS_ERR_ASM_X86_MNEMONICFAIL     EQU KS_ERR_ASM_MNEMONICFAIL
+
+; Runtime option for the Keystone engine ks_opt_type
+KS_OPT_SYNTAX                   EQU 1       ; Choose syntax for input assembly
+KS_OPT_SYM_RESOLVER             EQU 2       ; Set symbol resolver callback
+
+; Runtime option value (associated with ks_opt_type above) ks_opt_value
+KS_OPT_SYNTAX_INTEL             EQU 1       ; = 1 << 0, // X86 Intel syntax - default on X86 (KS_OPT_SYNTAX).
+KS_OPT_SYNTAX_ATT               EQU 2       ; = 1 << 1, // X86 ATT asm syntax (KS_OPT_SYNTAX).
+KS_OPT_SYNTAX_NASM              EQU 4       ; = 1 << 2, // X86 Nasm syntax (KS_OPT_SYNTAX).
+KS_OPT_SYNTAX_MASM              EQU 8       ; = 1 << 3, // X86 Masm syntax (KS_OPT_SYNTAX) - unsupported yet.
+KS_OPT_SYNTAX_GAS               EQU 16      ; = 1 << 4, // X86 GNU GAS syntax (KS_OPT_SYNTAX).
+KS_OPT_SYNTAX_RADIX16           EQU 32      ; = 1 << 5, // All immediates are in hex format (i.e 12 is 0x12)
+
+; Keystone Functions for x64 assemblers: JWasm/UASM
+ks_version                      PROTO :QWORD, :QWORD ; int* major, int* minor
+ks_arch_supported               PROTO :QWORD ; ks_arch arch
+ks_open                         PROTO :QWORD, :QWORD, :QWORD ; ks_arch arch, int mode, ks_engine** ks
+ks_close                        PROTO :QWORD ; ks_engine* ks
+ks_errno                        PROTO :QWORD ; ks_engine* ks
+ks_strerror                     PROTO :QWORD ; ks_err code
+ks_option                       PROTO :QWORD, :QWORD, :QWORD ; ks_engine* ks, ks_opt_type type, size_t value
+ks_asm                          PROTO :QWORD, :QWORD, :QWORD, :QWORD, :QWORD, :QWORD ; ks_engine* ks, const char* string, int64_t address, unsigned char** encoding, size_t* encoding_size, size_t* stat_count
+ks_free                         PROTO :QWORD ; unsigned char* p
+
+; Resolver callback to provide value for a missing symbol in @symbol.
+; To handle a symbol, the resolver must put value of the symbol in @value,
+; then returns True.
+; If we do not resolve a missing symbol, this function must return False.
+; In that case, ks_asm() would eventually return with error KS_ERR_ASM_SYMBOL_MISSING.
+;
+; To register the resolver, pass its function address to ks_option(), using
+; option KS_OPT_SYM_RESOLVER. typedef bool *ks_sym_resolver)(const char* symbol, uint64_t* value
+;
+; Invoke ks_option, ks, KS_OPT_SYM_RESOLVER, Addr ks_sym_resolver
+;
+
+ks_sym_resolver                 PROTO :QWORD, :QWORD ; const char* symbol, uint64_t* value
+
```

Comment:
can you remove these blank lines at the end of this file, and the other files?

Reply:
Yes, I have done that now.